### PR TITLE
update size and transitionProperty on horizontalChange

### LINKED ISSUE #33: Setting "horizontal" after element initialization breaks it
I tried to rebuild the demo in Dart Angular2 and the `#collapse2` doesn't resize when  `#collapse3` is opened.
I could reproduce this in the JS demo when I initially remove the `horizontal` attribute from `#collapse3` and then add it just before I expand it.
This way it has an redundant `height: 0px;` which breaks the demo.
It seems for the initial default "vertial" dimension the "height: 0px" is added but then when switched to "horizontal" it is not removed.

Looks like when using the element with Angular, the timing is somehow different which results in this behavior.

### PR DESCRIPTION
**Pull request by valdrinkoshi**
Fixes #33 by updating the size and transition property on horizontal changes.

### CODE REVIEW

**Comment by valdrinkoshi** [comment]
@notwaldorf @cdata kindly review

**Comment by notwaldorf** [comment]
Cool, LGTM

File: iron-collapse.html [hunk 1]
```diff
@@ -193,8 +193,13 @@
     },
 
     _horizontalChanged: function() {
+      // Reset old dimension.
+      this.style[this.dimension] = '';
```
Comment (notwaldorf) [inline iron-collapse.html]: Hmm, isn't `this.dimension` initially null, though?

Comment (valdrinkoshi) [inline iron-collapse.html]: good catch! will fix it

**Comment by valdrinkoshi** [comment]
@notwaldorf I've made `dimension` a read-only property, will be calculated each time is invoked.
PTAL

**Comment by notwaldorf** [comment]
i think this lgtm

**Pull request merged**